plugins/woocommerce/tests/legacy/unit-tests/rest-api/Tests/Version2/settings.php

```diff
@@ -482,29 +483,39 @@ public function test_classic_settings() {
 		$response = $this->server->dispatch( new WP_REST_Request( 'GET', '/wc/v2/settings/products' ) );
 		$data     = $response->get_data();
 		$this->assertTrue( is_array( $data ) );
-		$this->assertContains(
-			array(
-				'id'          => 'woocommerce_downloads_require_login',
-				'label'       => 'Access restriction',
-				'description' => 'Downloads require login',
-				'type'        => 'checkbox',
-				'default'     => 'no',
-				'tip'         => 'This setting does not apply to guest purchases.',
-				'value'       => 'no',
-				'_links'      => array(
-					'self'       => array(
-						array(
-							'href' => rest_url( '/wc/v2/settings/products/woocommerce_downloads_require_login' ),
+		$data_download_required_login = null;
+		foreach ( $data as $setting ) {
+			if ( 'woocommerce_downloads_require_login' === $setting['id'] ) {
+				$data_download_required_login = $setting;
+				break;
+			}
+		}
+		$this->assertNotEmpty( $data_download_required_login );
+		$this->assertEmpty(
```

Comment:
Would it be possible to use `assertEquals` here and other similar changes? It should be able to compare arrays properly.

Reply:
So we are not comparing the exact equality, what we want is that the response contains at least the fields that we have specified, but if it contains something more that's added by WordPress then it's not a problem. For example, now WP added `targetHints` but it could add something more in the future as well, and our tests won;t break again if it were to do so.

Reply:
Gotcha!